torchbenchmark/util/kernels/triton_fused_attention.py

```diff
@@ -508,4 +667,121 @@ def backward(ctx, do):
+        extra_kern_args = {}
+
+        TMA_SIZE = 128
+        BATCH, H, N_CTX = q.shape[0], q.shape[1], q.shape[2]
+        # no autotune with fixed BLOCK_N
+        '''
+        BLOCK_N = 128
+        desc_k = np.empty(TMA_SIZE, dtype=np.int8)
+        desc_v = np.empty(TMA_SIZE, dtype=np.int8)
+        # order is (0, 1) for fp8 in make_block_ptr, reverse here
+        triton.runtime.driver.active.utils.fill_2d_tma_descriptor(
+            k.data_ptr(),
+            BATCH * H * N_CTX,
+            HEAD_DIM_Q,
+            BLOCK_N,
+            HEAD_DIM_Q,
+            k.element_size(),
+            desc_k,
+        )
+        if v.dtype == torch.float8_e5m2:
+            triton.runtime.driver.active.utils.fill_2d_tma_descriptor(
+                v.data_ptr(),
+                BATCH * H * HEAD_DIM_Q,
+                N_CTX,
+                HEAD_DIM_Q,
+                BLOCK_N,
+                v.element_size(),
+                desc_v,
+            )
+        else:
+            triton.runtime.driver.active.utils.fill_2d_tma_descriptor(
+                v.data_ptr(),
+                BATCH * H * N_CTX,
+                HEAD_DIM_Q,
+                BLOCK_N,
+                HEAD_DIM_Q,
+                v.element_size(),
+                desc_v,
+            )
+        desc_k = torch.tensor(desc_k, device=v.device)
+        desc_v = torch.tensor(desc_v, device=v.device)
+        grid = lambda args: (triton.cdiv(q.shape[2], args["BLOCK_M"]), q.shape[0] * q.shape[1], 1)
+        '''
+        desc_k = torch.empty((TMA_SIZE), device="cuda", dtype=torch.int8)
+        desc_v = torch.empty((TMA_SIZE), device="cuda", dtype=torch.int8)
+        def grid_tma(META):
+            nonlocal desc_k
+            nonlocal desc_v
+            k_buf = torch.empty_like(desc_k, device="cpu")
+            v_buf = torch.empty_like(desc_v, device="cpu")
+            triton.runtime.driver.active.utils.fill_2d_tma_descriptor(
+                k.data_ptr(),
+                BATCH * H * N_CTX,
+                HEAD_DIM_Q,
+                META['BLOCK_N'],
+                HEAD_DIM_Q,
+                k.element_size(),
+                k_buf.numpy(),
+            )
+            if v.dtype == torch.float8_e5m2:
+                triton.runtime.driver.active.utils.fill_2d_tma_descriptor(
+                    v.data_ptr(),
+                    BATCH * H * HEAD_DIM_Q,
+                    N_CTX,
+                    HEAD_DIM_Q,
+                    META['BLOCK_N'],
+                    v.element_size(),
+                    v_buf.numpy(),
+                )
+            else:
+                triton.runtime.driver.active.utils.fill_2d_tma_descriptor(
+                    v.data_ptr(),
+                    BATCH * H * N_CTX,
+                    HEAD_DIM_Q,
+                    META['BLOCK_N'],
+                    HEAD_DIM_Q,
+                    v.element_size(),
+                    v_buf.numpy(),
+                )
+            desc_k.copy_(k_buf)
```

Comment:
After adding `pinned_memory=True` above, you can add `non_blocking=True` here to enable async memcpy.